modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/node/PatchNodeCommand.java

```diff
@@ -0,0 +1,153 @@
+
+            if ( updatedTargetNode != null )
+            {
+                patchedVersions.put( activeVersionMap.get( targetBranch ).getNodeVersionId(), updatedTargetNode.nodeVersionMetadata() );
+            }
+
+            result.addResult( targetBranch, updatedTargetNode != null ? updatedTargetNode.node() : null );
+        } );
+
+        return result.build();
+    }
+
+    private NodeVersionData patchNodeInBranch( final NodeId nodeId, final NodeVersionMetadata patchedVersionMetadata, final Branch branch )
+    {
+        final InternalContext internalContext = InternalContext.create( ContextAccessor.current() ).branch( branch ).build();
+
+        final Node persistedNode = nodeStorageService.get( nodeId, internalContext );
+
+        if ( persistedNode == null || !NodePermissionsResolver.hasPermission( internalContext.getPrincipalsKeys(), Permission.MODIFY,
+                                                                              persistedNode.getPermissions() ) )
+        {
+            return null;
+        }
+
+        if ( patchedVersionMetadata != null )
+        {
+            this.nodeStorageService.push( List.of( PushNodeEntry.create()
+                                                       .nodeBranchEntry( NodeBranchEntry.create()
+                                                                             .nodeVersionId( patchedVersionMetadata.getNodeVersionId() )
+                                                                             .nodePath( patchedVersionMetadata.getNodePath() )
+                                                                             .nodeVersionKey( patchedVersionMetadata.getNodeVersionKey() )
+                                                                             .nodeId( patchedVersionMetadata.getNodeId() )
+                                                                             .timestamp( patchedVersionMetadata.getTimestamp() )
+                                                                             .build() )
+                                                       .build() ), branch, l -> {
+            }, internalContext );
+
+            //TODO: refresh ?
+
+            return new NodeVersionData( nodeStorageService.get( nodeId, internalContext ), patchedVersionMetadata );
+        }
+        else
+        {
+            final EditableNode toBeEdited = new EditableNode( persistedNode );
+
+            params.getEditor().edit( toBeEdited );
+
+            final Node patchedNode = Node.create( toBeEdited.build() ).timestamp( Instant.now( CLOCK ) ).build();
+
+            return this.nodeStorageService.store( patchedNode, internalContext );
+        }
+    }
+
+    private Map<Branch, NodeVersionMetadata> getActiveNodeVersions( final NodeId nodeId, final Branches branches )
+    {
+        return params.getBranches() != null ? GetActiveNodeVersionsCommand.create( this )
```

Comment:
what is the case when branches are not specified?

Reply:
there are 3 options:
- upgrade the node in current branch
- do nothing
- throw an error

Currently it does nothing

Reply:
there is a 4th option: update in all